### PR TITLE
Switch to electron-builder for packaging

### PR DESCRIPTION
**Pull request by bmathews**
[`electron-builder`](https://github.com/electron-userland/electron-builder) wraps the packaging/installer dependencies we were using directly and should making signing releases on travis easier. I've tested the windows installer (win 10 only so far) and the .app/.dmg and all looks good. I've updated the README with instructions on how to use it. If someone else could give it all try, that'd be great!

/cc @mdaxtman @rgerstenberger

### CODE REVIEW

**Comment by rgerstenberger** [comment]
@bmathews can you explain a little more about how this gets rid of the `package.js` and replaces it with a different flow?

**Comment by bmathews** [comment]
@rgerstenberger Our `package.js`would manually run things like `electron-installer-dmg` and another file would run `electron-winstaller`, but `electron-builder` bundles all of that up into one package. So now, we just run the `.bin/build` that `electron-builder` provides, it reads the `/app/package.json` build property for configuration, and then everything Just Works™

**Comment by rgerstenberger** [comment]
Awesome, I'm testing this out locally right now.

File: README.md [hunk 1]
```diff
@@ -67,3 +67,24 @@ The slide list displays all slides in the presentation and allows for changing t
...
+```
+
+To build for Linux on MacOS, you'll need:
+```bash
+brew install gnu-tar libicns graphicsmagick xz
+```
+
+To build and build a package for all platforms, run:
+```bash
+npm run build && npm run package
```
Comment (rgerstenberger) [inline README.md]: Let's make this a single command. `npm release` or something like that.

File: package.json [hunk 2]
```diff
@@ -92,30 +74,32 @@
     "webpack-dev-middleware": "^1.6.1",
     "webpack-hot-middleware": "^2.10.0"
   },
-  "dependencies": {
```
Comment (rgerstenberger) [inline package.json]: Won't moving these to `app/package.json` break local dev for fresh clone + npm install? A dev would have to `cd app && npm install` correct? If so, we should add that to the readme.

Comment (rgerstenberger) [inline package.json]: ah, I see. THere's a postinstall script.

**Pull request merged**